nexus/reconfigurator/planning/src/blueprint_builder/builder.rs

```diff
@@ -715,12 +888,13 @@ impl<'a> BlueprintBuilder<'a> {
                 continue;
             };
 
-            let is_expunged =
-                is_already_expunged(zone_config, state).map_err(|error| {
-                    Error::Planner(anyhow!(error).context(format!(
-                        "for sled {sled_id}, error computing zones to expunge"
-                    )))
-                })?;
+            // TODO-john we lost the check for "are we expunging a zone we
```

Comment:
I"m not sure I really understand why that was there. If we mark a zone expunged in this iteration, why wouldn't we set the disposition to expunged in the editor?

Reply:
This check was something else; it was saying "if we're about to mark a zone as expunged, first check that we didn't already edit this zone in some way in this same planning pass". I think it was to guard against the planner adding a zone and then immediately expunging it, or changing the parameters of a zone and then expunging it, or other nonsense operations.

Reply:
Ah, that makes sense. I think it's probably fine to remove the check. We know when we want to expunge something which should trump everything else.